plugins/net_plugin/net_plugin.cpp

```diff
@@ -4597,7 +4597,6 @@ namespace eosio {
    // called from any thread
    void connections_manager::start_conn_timers() {
       start_conn_timer(connector_period, {}, timer_type::check); // this locks mutex
-      start_conn_timer(connector_period, {}, timer_type::stats); // this locks mutex
```

Comment:
If this line is removed, won't it prevent the prometheus `net_plugin` stats from being updated, even if the prometheus plugin is enabled? 
Unless I missed some other piece of code calling `connections_manager::connection_statistics_monitor`, but I didn't find any.
Maybe we should just have that function return immediately if `update_p2p_connection_metrics` is empty?

Reply:
See the next line below.

Reply:
Oh, I missed that, thanks Kevin.
I think it would be safer to check for `!!update_p2p_connection_metrics` inside `connection_statistics_monitor` rather than at the calling site, but I guess it works as is.